### PR TITLE
Group settings page

### PR DESCRIPTION
**Pull request by zachseidner1**
This pull request includes the following changes:

- Start on group settings, merge with main

- Get the frame of group settings ready, networking and implementations of a couple buttons are TODOs for Jonathan.

- Fix inconsistent date parsing functions

- Abstract group settings to separate component

- Fix text width issue and group settings issue

- Add leave group modal

### CODE REVIEW

File: src/pages/Group/GroupSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      navigate("/groups");
+    },
+    onError: () => {
+      setLeaveError("Error leaving group, check your internet connection.");
+    },
+  });
+  return (
+    <>
+      <Column className="mx-8 gap-6 pt-16">
+        <Row
```
Comment (jonathanjma) [inline src/pages/Group/GroupSettings.tsx:80]: Super minor but rn the cursor doesn't change when you hover over the back button. Can you make it so it changes to the click button cursor?

File: src/pages/Group/GroupSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      onError: () => {
+        setChangeGroupNameError(
+          "Error changing group name, check your internet connection.",
+        );
+      },
+    });
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  useEffect(() => {
+    if (!isFirstRender && groupNameText.trim() === "") {
+      setChangeGroupNameError("Group name cannot be empty.");
```
Comment (jonathanjma) [inline src/pages/Group/GroupSettings.tsx:58]: Hmm it still seems like I am able to set an empty group name?

File: src/pages/Group/GroupSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        <h4 className="text-gray-600">Manage Members</h4>
+        <Button label="View and Invite Members" variation="GRAY" />
+        <h4 className="text-gray-600">Leave Group</h4>
+        <Button
+          label="Leave Group"
+          variation="GRAY"
+          associatedModalId={confirmLeaveId}
+        />
+        <h4 className="text-gray-600">Delete Group</h4>
+        <Button label="Delete Group" variation="DANGEROUS" />
```
Comment (jonathanjma) [inline src/pages/Group/GroupSettings.tsx:115]: Were you supposed to implement delete group?

File: src/pages/Group/GroupSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  const [groupNameText, setGroupNameText] = useState("");
+  const [leaveError, setLeaveError] = useState("");
+  const [changeGroupNameError, setChangeGroupNameError] = useState("");
+  const { api } = useApi();
+  const queryClient = useQueryClient();
+
+  // leave modal
+  const confirmLeaveId = "confirm-leave";
+
+  const { mutate: saveGroupName, isLoading: putGroupNameLoading } =
```
Comment (jonathanjma) [inline src/pages/Group/GroupSettings.tsx:39]: something interesting I observed when changing group name, the order of the groups on the user groups page changes. Do you think you could just sort them by group id so the order stays the same?

**Review by jonathanjma — COMMENTED** [review]
Good work :-)

File: src/pages/Group/GroupSettings.tsx (hunk 1 above)
Comment (zachseidner1) [reply src/pages/Group/GroupSettings.tsx:80]: thanks for catching that!

**Review by jonathanjma — APPROVED** [review]

**Pull request merged**